lua/telescope/_extensions/bibtex.lua

```diff
@@ -0,0 +1,84 @@
+local has_telescope, telescope = pcall(require, "telescope")
+
+if not has_telescope then
+  error("This plugin requires telescope.nvim (https://github.com/nvim-telescope/telescope.nvim)")
+end
+
+local finders = require('telescope.finders')
+local pickers = require('telescope.pickers')
+local actions = require('telescope.actions')
+local previewers = require('telescope.previewers')
+local conf = require('telescope.config').values
+local scan = require('plenary.scandir')
+local path = require('plenary.path')
+
+local depth = 1
+
+local function read_file(file)
+  local entries = {}
+  local contents = {}
+  local p = path:new(file)
+  if not p:exists() then return {} end
+  local current_entry = "";
+  for line in p:iter() do
+    if line:match("@%w*{") then
+      local entry = line:gsub("@%w*{", "")
+      entry = entry:sub(1, -2)
+      current_entry = entry
+      table.insert(entries, entry)
+      contents[current_entry] = { line }
+    else
+      table.insert(contents[current_entry], line)
```

Comment:
This can leads to an issue. My first line is: `% Encoding: UTF-8`
So the else block is called first which leads to `contents[current_entry]` to be nil. Just do a check if nil then {} before inserting that line :)

Reply:
This and other special lines (i.e. "comments") now will not be added to the content. I detect when we are inside an entry and only add non-empty lines to the content of the entry. Lines before/after an entry will be ignored. I do this by checking for the closing `}` either in its own line or if there is one more `}` than `{` 
(I just recognized that this is the same case... and there might be issues with other multiline values like urls which need multiple brackets. I think in this case I will simply count all brackets in an entry. This should handle all cases).

Reply:
Seems to work now. You can always improve on it later.

Good job :)
Its feels weird to say that because its your repo but feel free to merge 🤣